src/main/java/io/dgraph/DgraphClient.java

```diff
@@ -28,9 +38,301 @@
  */
 public class DgraphClient {
   private static final String gRPC_AUTHORIZATION_HEADER_NAME = "authorization";
+  private static final String DGRAPH_SCHEME = "dgraph";
+  private static final String SSLMODE_DISABLE = "disable";
+  private static final String SSLMODE_REQUIRE = "require";
+  private static final String SSLMODE_VERIFY_CA = "verify-ca";
 
   private final DgraphAsyncClient asyncClient;
 
+  /**
+   * Options for configuring a Dgraph client connection.
+   *
+   * <p>Example use:
+   * <pre>{@code
+   * DgraphClient client = DgraphClient.ClientOptions.forAddress("localhost", 9080)
+   *     .withACLCredentials("username", "password")
+   *     .withTLS()
+   *     .build();
+   * }</pre>
+   */
+  public static class ClientOptions {
+    private final ManagedChannelBuilder<?> channelBuilder;
+    private String username;
+    private String password;
+    private String authorizationToken;
+    private final String host;
+    private final int port;
+
+    private ClientOptions(String host, int port) {
+      this.host = host;
+      this.port = port;
+      this.channelBuilder = ManagedChannelBuilder.forAddress(host, port);
+      // Default to plaintext
+      this.channelBuilder.usePlaintext();
+    }
+
+    /**
+     * Creates a new ClientOptions instance for the given host and port.
+     *
+     * @param host The hostname of the Dgraph server.
+     * @param port The port of the Dgraph server.
+     * @return A new ClientOptions instance.
+     */
+    public static ClientOptions forAddress(String host, int port) {
+      return new ClientOptions(host, port);
+    }
+
+    /**
+     * Sets username and password for ACL authentication.
+     *
+     * @param username The username for ACL authentication.
+     * @param password The password for ACL authentication.
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withACLCredentials(String username, String password) {
+      this.username = username;
+      this.password = password;
+      return this;
+    }
+
+    /**
+     * Sets an API key for authorization.
+     *
+     * @param apiKey The API key to use for authorization.
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withApiKey(String apiKey) {
+      this.authorizationToken = apiKey;
+      return this;
+    }
+
+    /**
+     * Sets a bearer token for authorization.
+     *
+     * @param token The bearer token to use for authorization.
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withBearerToken(String token) {
+      this.authorizationToken = "Bearer " + token;
+      return this;
+    }
+
+    /**
+     * Configures the client to use plaintext communication (no encryption).
+     *
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withPlaintext() {
```

Comment:
do we need this function when this is the default?

Reply:
It's for completeness and future-proofing.